tests/integration/dhcp_test.py

```diff
@@ -102,22 +155,47 @@ def test_ipv6_dhcp_only(dhcp_env):
     dhcp_cli_desired_state['ipv6']['enabled'] = True
     dhcp_cli_desired_state['ipv6']['dhcp'] = True
     dhcp_cli_desired_state['ipv6']['autoconf'] = False
+    dhcp_cli_desired_state['ipv6']['auto-routes'] = True
+    dhcp_cli_desired_state['ipv6']['auto-dns'] = True
+    dhcp_cli_desired_state['ipv6']['auto-gateway'] = True
 
     netapplier.apply(desired_state)
 
     assertlib.assert_state(desired_state)
+    time.sleep(5)  # wait dhcp
+    routes = _get_ipv6_main_routes()
+    for route in routes:
+        assert not ('default' in route and DHCP_CLI_NIC in route)
+        assert not (IPV6_CLASSLESS_ROUTE1 in route and DHCP_CLI_NIC in route)
+    assert DHCP_SRV_IP6 in _get_dns_conf()
 
 
 def test_ipv6_dhcp_and_autoconf(dhcp_env):
     desired_state = statelib.show_only((DHCP_CLI_NIC,))
     dhcp_cli_desired_state = desired_state[INTERFACES][0]
     dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
     dhcp_cli_desired_state['ipv6']['dhcp'] = True
     dhcp_cli_desired_state['ipv6']['autoconf'] = True
+    dhcp_cli_desired_state['ipv6']['auto-dns'] = True
+    dhcp_cli_desired_state['ipv6']['auto-gateway'] = True
+    dhcp_cli_desired_state['ipv6']['auto-routes'] = True
 
     netapplier.apply(desired_state)
 
     assertlib.assert_state(desired_state)
+    time.sleep(5)  # wait dhcp
+    routes = _get_ipv6_main_routes()
+    flag_has_gateway = False
+    flag_has_extra_route = False
+    for route in routes:
```

Comment:
Since this is used multiple times, it would be good to move this in a method, return `has_gateway` and `has_extra_route`, assert it to True here but also assert it to False when it should not be present. This makes it easier to understand what is happening.

Reply:
Moved to:

```python
    assert _has_ipv4_dhcp_gateway()
    assert _has_ipv4_classless_route()
```